### DISCUSSION TITLE
How to stop/start streams in bulk?

### DISCUSSION THREAD

**Discussion by Mohit-3196** [ant-media/Ant-Media-Server#5427] [FAQs]
I have over a hundred streams running on my Ant Media Server and I usually have to start/stop them individually.
Is there a way to stop/start all the streams at once?

Thank you

**Comment by USAMAWIZARD** [comment]
Currently there is no direct API to sop all the stream at once , But what you can do this by using this approach.
Here is how we can archive this 
1. Call API to Get all the streams in an application.
2. Call Stop / Start API in a loop to stop all the stream one by one.

we need to call management REST services to get all the streams which are available  in an application.for that we need to use JWT to access the APIs.

Management REST service requires security mechanism to call it. Alternatively, user can access to the REST methods of the application directly.

Here is how to  [Use JWT to access the resources for the management panel](https://antmedia.io/docs/guides/developer-sdk-and-api/rest-api-guide/management-rest-apis/#accessing-management-panel-rest-api-with-jwt-tokens)

https://antmedia.io/rest/ 


1.[ get all he running streams from the server side](https://antmedia.io/rest/#/ManagementRestService/getAppLiveStreams).
![image](https://github.com/ant-media/Ant-Media-Server/assets/47350008/f5bc9f4d-e66f-400b-908c-680b6a9590fa)

2. [stream stop API](https://antmedia.io/rest/#/BroadcastRestService/stopStreamingV2) in a  loop for all the streams in a shell script.
![image](https://github.com/ant-media/Ant-Media-Server/assets/47350008/5f0c99bf-2c40-4de2-8995-0d321d2c1406)

​
Call Stop API in a loop for all the streams.

**Reply by mekya** [reply]
Hi @USAMAWIZARD,

Thank you for the answer. No need to access to the resources with username and password. 

- [Use JWT to access the resources for the management panel](https://antmedia.io/docs/guides/developer-sdk-and-api/rest-api-guide/management-rest-apis/#accessing-management-panel-rest-api-with-jwt-tokens)

- Alternatively, user can access to the REST methods of the application directly 
https://antmedia.io/rest/

**Reply by USAMAWIZARD** [reply]
Thanks let me update my answer
